rfcs/0004-pablo-distribution.md

```diff
@@ -0,0 +1,447 @@
+    -   [2.2. Pool Fees](#_pool_fees)
+        -   [2.2.1. LP Fee Distribution](#_lp_fee_distribution)
+-   [3. Use Cases](#_use_cases)
+-   [4. Requirements](#_requirements)
+    -   [4.1. Pablo Liquidity Providers](#_pablo_liquidity_providers)
+    -   [4.2. PBLO Stakers](#_pblo_stakers)
+    -   [4.3. PICA Stakers](#_pica_stakers)
+    -   [4.4. Pablo Governance](#_pablo_governance)
+    -   [4.5. PICA Governance](#_pica_governance)
+    -   [4.6. Technical Requirements](#_technical_requirements)
+-   [5. Method](#_method)
+    -   [5.1. System Overview](#_system_overview)
+    -   [5.2. Pallet-Pablo](#_pallet_pablo)
+        -   [5.2.1. FeeConfig](#_feeconfig)
+        -   [5.2.2. LP Trading Fee
+            Distribution](#_lp_trading_fee_distribution)
+            -   [5.2.2.1. LP fNFT
+                Configuration](#_lp_fnft_configuration)
+            -   [5.2.2.2. LP fNFT Creation When Receiving LP Tokens for
+                Adding
+                Liquidity](#_lp_fnft_creation_when_receiving_lp_tokens_for_adding_liquidity)
+            -   [5.2.2.3. LP fNFT Resizing or Unstaking When Burning LP
+                Tokens for Removing
+                Liquidity](#_lp_fnft_resizing_or_unstaking_when_burning_lp_tokens_for_removing_liquidity)
+            -   [5.2.2.4. Trading Fee
+                Distribution](#_trading_fee_distribution)
+        -   [5.2.3. PBLO Staker Trading Fee
+            Distribution](#_pblo_staker_trading_fee_distribution)
+    -   [5.3. Pallet Staking Rewards](#_pallet_staking_rewards)
+        -   [5.3.1. Staking Reward
+            Calculation](#_staking_reward_calculation)
+        -   [5.3.2. PICA/PBLO Token Staking](#_picapblo_token_staking)
+            -   [5.3.2.1. RewardConfig](#_rewardconfig)
+            -   [5.3.2.2. Governance Sets the RewardConfig and Transfers
+                the Reward
+                Allocation](#_governance_sets_the_rewardconfig_and_transfers_the_reward_allocation)
+            -   [5.3.2.3. Governance Sets the Pablo Pool LP Staking
+                Reward
+                Allocation](#_governance_sets_the_pablo_pool_lp_staking_reward_allocation)
+            -   [5.3.2.4.
+                RewardAccumulationHook](#_rewardaccumulationhook)
+-   [6. Implementation](#_implementation)
+    -   [6.1. Pallet Pablo: LP Fee + Staking
+        Changes](#_pallet_pablo_lp_fee_staking_changes)
+    -   [6.2. Pallet Staking Rewards: PICA/PBLO Staking Related
+        Changes](#_pallet_staking_rewards_picapblo_staking_related_changes)
+-   [Appendix A: Fee Distribution Q&A](#_fee_distribution_qa)
+
+## 1. Abstract
+
+This document proposes the Pablo distribution(token and pool trading
+fees) mechanism while considering various options and capturing the
+discussions about the subject.
+
+`TODO summarise the mechanism`
+
+## 2. Background
+
+### 2.1. PBLO Token Initial Distribution
+
+According to the [tokenomics design](http://link) the Pablo is supposed
```

Comment:
is it broken?

Reply:
There is a design though I have not received the link yet, hence it's kind of a to do